employer_recommendation_system/emp/views.py

```diff
@@ -157,6 +157,13 @@ def student_homepage(request):
     context['jobs_to_display'] = jobs_to_display if len(jobs_to_display)<6 else jobs_to_display[:6]
     rec_jobs = get_recommended_jobs(rec_student)
     context['rec_jobs'] = rec_jobs if len(applied_jobs)<3 else rec_jobs[:3]
+    active_event = Event.objects.filter(status=1,type='JOBFAIR').order_by('-start_date').first()
+    is_eligible = TestAttendance.objects.filter(student_id=rec_student.spk_student_id,mdlgrade__gte=settings.PASS_GRADE).exists()
+    context['is_eligible'] = is_eligible
+    if active_event and is_eligible :
+        context['active_event'] = active_event
+        context['jobfair'] = JobFair.objects.get(event=active_event)
+        context['is_registered_for_jobfair'] = JobFair.objects.filter(event=active_event,students=rec_student).exists()
```

Comment:
Not sure what is happening here.

JobFair.objects.get(event=active_event) gets one object only.

What does JobFair.objects.filter(event=active_event,students=rec_student) return? I feel it will return the same object in QuerySet form.

What is the intention of this code?